code/modules/interface/mob_input.dm

```diff
@@ -125,6 +126,20 @@
 				unpull_particle(src,pulling)
 			src.pulling = null
 
+/**
+  *
+	* Return the ability bound to the pressed ability hotkey combination
+	*
+  */
+/mob/proc/get_ability_hotkey(mob/user, parameters)
+	if(!parameters["left"]) return
+	if (parameters["ctrl"] && user?.abilityHolder?.ctrlPower)
+		return user.abilityHolder.ctrlPower
+	if (parameters["alt"] && user?.abilityHolder?.altPower)
+		return user.abilityHolder.altPower
+	if (parameters["shift"] && user?.abilityHolder?.shiftPower)
+		return user.abilityHolder.shiftPower
```

Comment:
probably best to just check if user.abilityholder checks before all these, just simpler than all the `?.`

Reply:
```suggestion
	if(!user?.abilityHolder) return
	if(parameters["ctrl"] && user.abilityHolder.ctrlPower)
		return user.abilityHolder.ctrlPower
	if(parameters["alt"] && user.abilityHolder.altPower)
		return user.abilityHolder.altPower
	if(parameters["shift"] && user.abilityHolder.shiftPower)
		return user.abilityHolder.shiftPower
```